### DISCUSSION TITLE
SD.Next Release 12-11-2025

### DISCUSSION THREAD

**Discussion by vladmandic** [vladmandic/sdnext#4464] [Announcements]
## SD.Next release 2025-12-11

*What's new?*  
New native [kanvas](https://vladmandic.github.io/sdnext-docs/Kanvas/) module for image manipulation that fully replaces *img2img*, *inpaint* and *outpaint* controls, massive update to **Captioning/VQA** models and features  
New generation of **Flux.2** large image model, new **Z-Image** model that is creating a lot of buzz, new **Kandinsky 5 Lite** image model and new **Photoroom PRX** model  
And first cloud models with **Google Nano Banana** *2.5 Flash and 3.0 Pro* and **Google Veo** *3.1* video model  
Also new are **HunyuanVideo 1.5** and **Kandinsky 5 Pro** video models  
Plus a lot of internal improvements and fixes  

![Screenshot](https://github.com/user-attachments/assets/54b25586-b611-4d70-a28f-ee3360944034)

[ReadMe](https://github.com/vladmandic/automatic/blob/master/README.md) | [ChangeLog](https://github.com/vladmandic/automatic/blob/master/CHANGELOG.md) | [Docs](https://vladmandic.github.io/sdnext-docs/) | [WiKi](https://github.com/vladmandic/automatic/wiki) | [Discord](https://discord.com/invite/sd-next-federal-batch-inspectors-1101998836328697867) | [Sponsor](https://github.com/sponsors/vladmandic)  

### Details for 2025-12-11

- **Models**
  - [Black Forest Labs FLUX.2 Dev](https://bfl.ai/blog/flux-2) and prequantized variation [SDNQ-SVD-Uint4](https://huggingface.co/Disty0/FLUX.2-dev-SDNQ-uint4-svd-r32)  
    **FLUX.2-Dev** is a brand new model from BFL and uses large 32B DiT together with Mistral 24B as text encoder  
    model is available for text, image and edit tasks and can optionally use control input as second input image  
    this is a very large model at ~100GB, so use of prequantized model at ~32GB is strongly advised
    using prequant version and default offloading, model runs on GPUs with ~20GB  
    *note*: model is [gated](https://vladmandic.github.io/sdnext-docs/Gated/)  
  - [Z-Image Turbo](https://huggingface.co/Tongyi-MAI/Z-Image-Turbo) and prequantized variation [SDNQ-SVD-Uint4](https://huggingface.co/Disty0/Z-Image-Turbo-SDNQ-uint4-svd-r32)  
    **Z-Image** is a powerful and highly efficient image generation model with 6B parameters and using Qwen-3 as text encoder  
    unlike most of new models that are far larger, Z-Image architecture allows it to run with good performance even on mid-range hardware  
    *note*: initial release is *Turbo* variant only with *Base* and *Edit* variants to follow  
  - [Kandinsky 5.0 Lite]() is a new 6B model using Qwen-2.5 as text encoder  
    it comes in text-to-image and image-edit variants  
  - **Google Gemini Nano Banana** [2.5 Flash](https://blog.google/products/gemini/gemini-nano-banana-examples/) and [3.0 Pro](https://deepmind.google/models/gemini-image/pro/) 
    first cloud-based model directly supported in SD.Next UI  
    *note*: need to set `GOOGLE_API_KEY` environment variable with your key to use this model  
  - [Photoroom PRX 1024 Beta](https://huggingface.co/Photoroom/prx-1024-t2i-beta)  
    PRX (Photoroom Experimental) is a small 1.3B parameter t2i model trained entirely from scratch, it uses T5-Gemma text-encoder  
- **Video**  
  - [HunyuanVideo 1.5](https://huggingface.co/tencent/HunyuanVideo-1.5) in T2V and I2V variants, both standard and distilled and both 720p and 480p resolutions  
    **HunyuanVideo 1.5** improves upon previous 1.0 version with better quality and higher resolution outputs, it uses Qwen2.5-VL text-encoder  
    distilled variants provide faster generation with slightly reduced quality  
  - [Kandinsky 5.0 Pro Video](https://huggingface.co/kandinskylab/Kandinsky-5.0-T2V-Pro-sft-5s-Diffusers) in T2V and I2V variants  
    larger 19B (and more powerful version) of previously released Lite 2B models  
  - [Google Veo 3.1](https://gemini.google/us/overview/video-generation/) for T2V and I2V workflows  
    *note*: need to set `GOOGLE_API_KEY` environment variable with your key to use this model  
- **Kanvas**: new module for native canvas-based image manipulation  
  kanvas is a full replacement for *img2img, inpaint and outpaint* controls  
  see [docs](https://vladmandic.github.io/sdnext-docs/Kanvas/) for details  
  *experimental*: report any feedback in master [issue](https://github.com/vladmandic/sdnext/issues/4358)  
- **Captioning** and **VQA: Visual Question & Answer**  
  massive update to both features and supported models, thanks @CalamitousFelicitousness  
  models:  
  - additional `mooondream-2` features  
  - support for `moondream-3-preview`  
  - support for `qwen3-vl` with thinking  
  - additional `gemma-3-vl` finetunes  
  - support for `XiaomiMiMo`  
  ui:  
  - ability to annotate actual image, not just generate captions/answers  
    e.g. actualy mark detected regions/points  
  features:  
  - ui indicator of model capabilities  
  - support for *prefill* style of prompting/answering  
  - support for *reasoning* mode for supported models  
    with option to output answer-only or reasoning-process   
  - additional debug logging  
- **Other Features**
  - **wildcards**: allow recursive inline wildcards using curly braces syntax  
  - **sdnq**: simplify pre-quantization saved config  
  - **attention**: additional torch attention settings  
  - **lora**: separate fuse setting for native-vs-diffuser implementations  
  - **auth**: strong-enforce auth check on all api endpoints  
  - **amdgpu**: prefer rocm-on-windows over zluda  
  - **amdgpu**: improve rocm-on-windows installer  
  - **sdnq**: improve dequant logic  
  - **gallery**: significant performance improvements, thanks @awsr  
- **API**
  - `/control` endpoint is now fully compatible with scripts  
  - `/control` additional params to to control *xyz grid*  
    see `cli/api-xyz.py` for simple example  
  - `/detailers` new endpoint to list available detailers, both built-in and any custom downloaded  
  - `/face-restorers` expanded to list model folders  
- **Internal**
  - python: set 3.10 as minimum supported version  
  - sdnq: multiple improvements to quantization and dequantization logic
  - torch: update to `torch==2.9.1` for *cuda, ipex, openvino, rocm* backends
  - attention: refactor attention handling  
  - scripts: remove obsolete video scripts  
  - lint: update global lint rules  
  - chrono: switch to official pipeline  
  - pipeline: add optional preprocess and postprocess hooks  
  - auth: wrap all internal api calls with auth check and use token when possible  
  - installer: reduce requirements  
  - installer: auto-restart on self-update  
  - server: set correct mime-types  
  - sdnq: unconditional register on startup  
  - python: start work on future-proofing for modern python versions, thanks @awsr  
  - nunchaku: update to `1.0.2`  
  - lint: add rules for run-on-windows  
  - gallery: setting to enable/disable client-side caching, thanks @awsr  
  - gallery: faster thumbnail generation, thanks @awsr  
  - gallery: purge old thumbnails, thanks @awsr  
- **Docs**
  - update supported models table with VAE information, thanks @alerikaisattera
- **Fixes**
  - xyz-grid: improve parsing of axis lists, thanks @awsr  
  - hires: strength save/load in metadata, thanks @awsr  
  - imgi2img: fix initial scale tab, thanks @awsr  
  - img2img: fix restoring refine sampler from metadata, thanks @awsr
  - log: client log formatting, thanks @awsr  
  - rocm: check if installed before forcing install  
  - pony-v7: fix text-encoder  
  - detailer: with face-restorers  
  - detailer: using lora in detailer prompt  
  - detailer: fail on unsupported models instead of corrputing results  
  - ui: fix collapsible panels  
  - svd: fix stable-video-diffusion dtype mismatch  
  - animatediff: disable sdnq if used  
  - lora: restore pipeline type if reload/recompile needed  
  - process: improve send-to functionality  
  - control: safe load non-sparse controlnet  
  - control: fix marigold preprocessor with bfloat16  
  - auth: fix password being shown in clear text during login  
  - firefox: remove obsolete checks, thanks @awsr  
  - runai streamer: cleanup logging, thanks @CalamitousFelicitousness  
  - gradio: event handlers, thanks @awsr
